File: KuantumKaosYonetimi/CSharp/KuantumKaosYonetimi/KaranlikMadde.cs
namespace KuantumKaosYonetimi
{
    // KaranlikMadde: KuantumNesnesi'nden türemiş ve IKritik arayüzünü uygulamıştır.
    public class KaranlikMadde : KuantumNesnesi, IKritik
    {
        public KaranlikMadde(string id, double stabilite)
            : base(id, 8, stabilite) // TehlikeSeviyesi: 8 (Yüksek)
        {
        }

        // AnalizEt() (Çok Biçimlilik/Polymorphism)
        public override void AnalizEt()
        {
            // Stabilite 15 birim düşer.
            Stabilite -= 15;
            Console.WriteLine($"[ID: {ID}] Karanlık Madde analizi yapıldı. Stabilite -15. Yeni Stabilite: {Stabilite:F2}%");
        }

        // AcilDurumSogutmasi() (IKritik Uygulaması)
        public void AcilDurumSogutmasi()
        {
            // Stabiliteyi +50 artırır (Kapsülleme max 100'ü halleder).
            Stabilite += 50;
            Console.WriteLine($"[ID: {ID}] Karanlık Madde soğutuldu! Stabilite +50. Yeni Stabilite: {Stabilite:F2}%");
        }
    }
}

File: KuantumKaosYonetimi/Javascript/KuantumKaosYonetimi/KaranlikMadde.js
// KaranlikMadde.js
const KuantumNesnesi = require('./KuantumNesnesi');
// IKritik'i require etmeye gerek yoktur, sadece metodu implemente ederiz.

class KaranlikMadde extends KuantumNesnesi {
    constructor(id, stabilite) {
        super(id, 8, stabilite); // TehlikeSeviyesi: 8
    }

    // AnalizEt() (Çok Biçimlilik/Polymorphism)
    AnalizEt() {
        this.Stabilite -= 15;
        console.log(`[ID: ${this.ID}] Karanlık Madde analizi yapıldı. Stabilite -15. Yeni Stabilite: ${this.Stabilite.toFixed(2)}%`);
    }

    // AcilDurumSogutmasi() (IKritik Uygulaması)
    AcilDurumSogutmasi() {
        this.Stabilite += 50; // Setter max 100'ü halleder.
        console.log(`[ID: ${this.ID}] Karanlık Madde soğutuldu! Stabilite +50. Yeni Stabilite: ${this.Stabilite.toFixed(2)}%`);
    }
}

module.exports = KaranlikMadde;